### ISSUE TITLE
Cargo wagons don't spawn an exit

### ISSUE THREAD

**Issue by TheWiznard** [Gerkiz/Immersive_Locomotives#6]
Immersive Locomotives - Version: 2.0.0
Factorio 2.0.20 Space Age

Description of the issue: When I enter a cargo wagon by itself, no attached train or wagon it does not spawn the cars on the sides that lets you leave, thus trapping you inside the cargo wagon. 

I tried before an after researching the trains technology, but the car to exit still did not spawn. Entering a train with no wagons attached still spawned the exits. When a single wagon is attached to the train I can enter the train and the exits appear on the train but there are no exits on the attached wagon. Also I'm not sure since its my first time trying the mod but I also could not see more than 1 wagon from entering the train, for example I tried train, wagon, wagon ; and only saw the the first wagon space. 

It might also not be a problem with the mod and might be one of the other mods I'm using so I'm including my modlist here for reference.
```
base
elevated-rails
quality
space-age
aai-containers
agricultural-tower-placement-helper
alien-biomes
alien-biomes-graphics
ArmouredBiters
AutoDeconstruct
automatic-discharge-defense
behemoth-enemies
belt-visualizer
bobinserters
brake-squeal-modded
cargo-ships
cargo-ships-graphics
CleanFloor
Cold_biters
configurable-valves
Constructron-Continued
deadlock-beltboxes-loaders
DiscoScience
distant-misfires
diurnal-dynamics
drawing-board
dredgeworks
dynamic-robot-queues
Electric_flying_enemies
Enduring-Embers
EpicArtillerySounds
equipment-gantry
ev-refining
even-distribution
Explosive_biters
factorissimo-2-notnotmelon
fire-lights
flamethrower-wagon
flib
flow-config
Fridge
FrozenFood
ghost-counter
GUI_Unifyer_Unified
Hidden_Loot
ice-actually-melts
Immersive_Locomotives
IntermodalContainers
inventory-repair
LargerLamps-2_0
lawn-mower
LogiNetChannels
LouderCargoWagons
Milestones
Mini_Trains
mushroom-cloud
naufulglebunusilo
no--inventory-is-full--corpse-pickup-troubles
notice-me-senpai
no_wall_alerts
PLORD_Prometheus_GrenadeLauncher
puppy-vehicle-fuel-ui
RateCalculator
rec-blue-plus
RPGsystem
science-pack-glow-reborn
SearchlightAssault
shield-generators
signal-display
skip-hours
Spiderissmo
StableFoundations
Stacked_Mining
StatsGui
terrapalus
textplates
toolbars-mod
Toxic_biters
vehicle-corpses
vehicle_physics
vtk-armor-reach
zithorian-extra-storage-tanks-port
```

**Comment by TheWiznard** [comment]
my bad! 😅 tested it myself with no other mods and works as intended, besides my problems have been enjoying it so far! thanks

**Comment by TheWiznard** [comment]
ok so further testing I confirmed what was going on, if any future google bros search this it was kinda obvious lol but trains tech has to be researched as per the mod description. if you place a wagon before you research the tech the exits don't spawn. What I didn't test initially was placing a new train and wagon after I researched trains. after placing a completely new train and wagon the exits spawn as they should still not sure about linked chests but i'm figuring that out as I go.
